File: app/src/main/java/com/samrat/voicealert/SmsReceiver.java
package com.samrat.voicealert;
import android.content.*; import android.provider.Telephony;
public class SmsReceiver extends BroadcastReceiver {
public void onReceive(Context c, Intent i){
for(var sms: Telephony.Sms.Intents.getMessagesFromIntent(i)){
String b=sms.getDisplayMessageBody(); double amt=MainActivity.extractAmount(b);
if(amt>0 && MainActivity.instance!=null){ boolean cr=MainActivity.isCredit(b);
MainActivity.instance.runOnUiThread(()->MainActivity.instance.showAlert(amt,cr,"bank")); }
}
}
}

File: app/src/main/java/com/samrat/voicealert/SmsReceiver.kt
package com.samrat.voicealert
import android.content.*
import android.provider.Telephony
class SmsReceiver : BroadcastReceiver() {
    override fun onReceive(c: Context, i: Intent) {
        val msgs = Telephony.Sms.Intents.getMessagesFromIntent(i)
        val body = msgs.joinToString("") { it.messageBody }
        if (!body.contains("credited", true) && !body.contains("UPI", true)) return
        if (body.contains("debited", true)) return
        val amt = Regex("""Rs\.?\s?([\d,]+\.?\d*)""").find(body)?.groupValues?.get(1)?.replace(",","")?.toDoubleOrNull() ?: return
        TtsHelper.speakStatic(c, amt)
    }
}
